apps/api/v1/pages/api/event-types/[id]/_delete.ts

```diff
@@ -49,9 +51,18 @@ async function checkPermissions(req: NextApiRequest) {
   const { userId, isSystemWideAdmin } = req;
   const { id } = schemaQueryIdParseInt.parse(req.query);
   if (isSystemWideAdmin) return;
-  /** Only event type owners can delete it */
-  const eventType = await prisma.eventType.findFirst({ where: { id, userId } });
+
+  const eventType = await prisma.eventType.findFirst({ where: { id } });
+
   if (!eventType) throw new HttpError({ statusCode: 403, message: "Forbidden" });
+
+  /** Only event type owners or team owners for team events can delete it */
+  if (eventType.teamId) await checkTeamEventEditPermission(req, eventType);
+
+  if (eventType.parentId) await checkParentEventOwnership(req);
```

Comment:
```suggestion
  if (eventType.teamId) return await checkTeamEventEditPermission(req, eventType);

  if (eventType.parentId) return await checkParentEventOwnership(req);
```

Minor NIT: can add early returns